Runtime/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/Util.java

```diff
@@ -63,4 +73,48 @@
       return null;
     }
   }
+
+  /**
+   * Check if the application is debuggable.
+   *
+   * This method check if the application is signed by a debug key.
+   * https://stackoverflow.com/questions/7085644/how-to-check-if-apk-is-signed-or-debug-build
+   *
+   * @param context application context
+   * @return true if app is debuggable, false otherwise
+   */
+  static boolean isAppDebuggable(Context context)
+  {
+    boolean debuggable = false;
+
+    try
+    {
+      PackageInfo pinfo = context.getPackageManager().getPackageInfo(context.getPackageName(), PackageManager.GET_SIGNATURES);
+      Signature signatures[] = pinfo.signatures;
+
+      CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+      for ( int i = 0; i < signatures.length;i++)
+      {
+        ByteArrayInputStream stream = new ByteArrayInputStream(signatures[i].toByteArray());
+        X509Certificate cert = (X509Certificate) cf.generateCertificate(stream);
+        debuggable = cert.getSubjectX500Principal().equals(DEBUG_DN);
```

Comment:
just a general note: `DEBUG_DN` will never be null, `cert.getSubjectX500Principal()` could be, at least there is no non-null contract on the method. so i would prefer this:
```diff
- debuggable = cert.getSubjectX500Principal().equals(DEBUG_DN);
+ debuggable = DEBUG_DN.equals(cert.getSubjectX500Principal());
```